apps/chatbot/src/app/routers/queries.py

```diff
@@ -0,0 +1,162 @@
+import nh3
+import os
+import uuid
+import json
+import yaml
+import requests
+from botocore.exceptions import BotoCoreError, ClientError
+from boto3.dynamodb.conditions import Key
+from fastapi import APIRouter, Header, HTTPException
+from logging import getLogger
+from typing import Annotated
+from src.app.models import Query, tables
+from src.app.sessions import (
+    current_user_id,
+    find_or_create_session,
+    session_salt,
+    hash_func,
+    last_session_id,
+    get_user_session
+)
+from src.modules.chatbot import Chatbot
+
+EVALUATION_API_URL = os.getenv("CHB_EVALUATION_API_URL", "http://api:8080/evaluations")
+
+logger = getLogger(__name__)
+router = APIRouter()
+params = yaml.safe_load(open("config/params.yaml", "r"))
+prompts = yaml.safe_load(open("config/prompts.yaml", "r"))
+chatbot = Chatbot(params, prompts)
+
+
+def evaluate(evaluation_data: dict, authorization: str) -> dict:
+    # TODO: make it async
+    evaluation_result = {}
+    EVALUATION_USE_API = os.getenv("CHB_EVALUATION_USE_API", "false")
+    if EVALUATION_USE_API == "false":
+        logger.info("[queries.evaluate] call chatbot.evaluate...") 
+        evaluation_result = chatbot.evaluate(
+            query_str=evaluation_data["query_str"],
+            response_str=evaluation_data["answer"],
+            retrieved_contexts=evaluation_data["retrived_context"],
+            trace_id=evaluation_data["trace_id"],
+            session_id=evaluation_data["session_id"],
+            user_id=evaluation_data["user_id"],
+            messages=evaluation_data["messages"]
+        )
+    else:
+        logger.info(
+            f"[queries.evaluate] call requests.post {EVALUATION_API_URL}..."
+        )
+        evaluation_result = requests.post(
+            EVALUATION_API_URL,
+            data=json.dumps(evaluation_data),
+            headers={
+                "Authorization": authorization,
+                "Content-Type": "application/json"
+            }
+        )
+
+    logger.info(f" ------>>> [queries] evaluation_result={evaluation_result.json()})")
+    return evaluation_result.json()
+
+
+@router.post("/queries")
+async def query_creation(
+    query: Query,
+    authorization: Annotated[str | None, Header()] = None
+):
+    now = datetime.datetime.now(datetime.UTC)
+    trace_id = str(uuid.uuid4())
+    userId = current_user_id(authorization)
+    session = find_or_create_session(userId, now=now)
+    salt = session_salt(session['id'])
+    query_str = nh3.clean(query.question)
+    user_id = hash_func(userId, salt)
+    messages = (
+        [item.dict() for item in query.history] if query.history else None
+    )
+
+    logger.info(f" ------>>> [queries] call chat_generate(query_str={query_str})")
+    answer, retrived_context = chatbot.chat_generate(
+        query_str=query_str,
+        trace_id=trace_id,
+        session_id=session["id"],
+        user_id=user_id,
+        messages=messages,
+    )
+
+    # TODO: add langfuse to compose.test.yaml
```

Comment:
```suggestion
    # TODO: add langfuse to compose.test.yaml
```
I think you should add a task in backlog and remove this TODO